src/lib/xmlparse.rs

```diff
@@ -4476,14 +4479,10 @@ impl<'scf> XML_ParserStruct<'scf> {
                     j_0 = 0;
                     while j_0 < (*b).uriLen {
                         let c: XML_Char = *(*b).uri.offset(j_0 as isize);
-                        if if self.m_tempPool.ptr == self.m_tempPool.end as *mut XML_Char
-                            && !self.m_tempPool.grow()
-                        {
+                        if if self.m_tempPool.is_full() && !self.m_tempPool.grow() {
```

Comment:
I managed to remove all external `grow` calls except one in `storeEntityValue` where it starts by trying to grow the pool. Not sure if it's a good idea to remove. We do call append_char on it much later but removing it could change the control flow by not returning early on OOM